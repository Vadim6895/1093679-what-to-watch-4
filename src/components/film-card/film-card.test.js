import React from "react";
import renderer from "react-test-renderer";
import FilmCard from "./film-card.jsx";

const filmCards = [{
  movieName: `The Grand Budapest Hotel`,
  productionDate: `1984`,
  genre: `Drama`,
  moviePoster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  id: 0
}];

it(`Should FilmCard render correctly`, () => {
  const tree = renderer
  .create(<FilmCard
    name={filmCards[0].movieName}
    id={filmCards[0].id}
    moviePoster={filmCards[0].moviePoster}
    onFilmClick={() => {}}
    onMouseEnter={() => {}}
    onMouseLeave={() => {}}
    src={filmCards[0].src}
    isPlaying={false}
  />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
