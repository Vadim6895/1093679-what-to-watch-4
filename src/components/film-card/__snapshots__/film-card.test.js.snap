// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmCard render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
>
  <video
    className="small-movie-player"
    height="175"
    muted={true}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    width="280"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
    >
      The Grand Budapest Hotel
    </a>
  </h3>
</article>
`;
